lerobot/common/robot_devices/cameras/opencv.py

```diff
@@ -347,43 +393,45 @@ def async_read(self):
             )
 
         if self.thread is None:
-            self.stop_event = threading.Event()
-            self.thread = Thread(target=self.read_loop, args=())
-            self.thread.daemon = True
-            self.thread.start()
+            with self.lock:
+                self.stop_event = threading.Event()
+                self.thread = Thread(target=self.read_loop, args=())
+                self.thread.daemon = True
+                self.thread.start()
 
         num_tries = 0
-        while self.color_image is None:
-            num_tries += 1
-            time.sleep(1 / self.fps)
-            if num_tries > self.fps and (self.thread.ident is None or not self.thread.is_alive()):
-                raise Exception(
-                    "The thread responsible for `self.async_read()` took too much time to start. There might be an issue. Verify that `self.thread.start()` has been called."
-                )
+        while True:
+            # Do not use `with self.lock` here, as it reduces fps
+            if self.color_image is not None:
+                return self.color_image
```

Comment:
@aliberts `with self.lock` causing slow fps here

Before fix:
```python
with self.lock:
    if self.color_image is not None:
        return self.color_image
```
<img width="1448" alt="Screenshot 2024-09-28 at 15 38 19" src="https://github.com/user-attachments/assets/eafc77ce-94d0-42a1-bd65-56c534fdf384">

After fix:
```python
if self.color_image is not None:
    return self.color_image
```
<img width="1448" alt="Screenshot 2024-09-28 at 15 42 29" src="https://github.com/user-attachments/assets/69772ba8-1953-43f5-a73c-43b2a01fe048">

Reply:
It shouldn't be necessary to `lock` for a read-only device (unless you're specifically worried about an image being over-written in-place as you're reading it)
-- something that shouldn't usually be an issue even if it did happen. 
It may make this function call non-deterministic (inconsistent behavior between runs) but that was already understood to be the case for an `async` function.

Reply:
Removed the `self.lock`